kong/db/dao/plugins.lua

```diff
@@ -234,14 +234,18 @@ local function load_plugin(self, plugin)
     return nil, err
   end
 
-  if schema.fields.consumer and schema.fields.consumer.eq == null then
-    plugin.no_consumer = true
-  end
-  if schema.fields.route and schema.fields.route.eq == null then
-    plugin.no_route = true
-  end
-  if schema.fields.service and schema.fields.service.eq == null then
-    plugin.no_service = true
+  for _, field in ipairs(schema.fields) do
+    if field.consumer and field.consumer.eq == null then
+      handler.no_consumer = true
```

Comment:
Is this intentionally changed to `handler.no_consumer` from `plugin.no_consumer`?

Reply:
Good question. Yes, note that the function returns `handler` rather than `plugin`, and that is what is used in places where the function is called.

Reply:
Yes, but it seems that in https://github.com/Kong/kong/blob/master/kong/runloop/plugins_iterator.lua#L134-L142 where the values are used, it is checking the `plugin` object instead of the handler.

Reply:
That's very right. Although `plugin` in here in the DAO is the plugin name, `plugin` over in the plugins iterator is an object with a `handler` field. Updated to fix it. Good catch!